### ISSUE TITLE
[TF 2.0] tf.summary step argument should cast to int64

### ISSUE THREAD

**Issue by nfelt** [tensorflow/tensorflow#26407]
The TF 2.0 versions of tf.summary.scalar(), image(), etc now require a `step` argument at the callsite. This step argument should allow at least any integral type and probably also floating point types (at least if they have no fractional part), but it currently requires a python integer or a `tf.int64` tensor.

This results in errors like:

```python
tf.summary.scalar('foo', 22, step=tf.constant(1))
...
ValueError: Tensor conversion requested dtype int64 for Tensor with dtype int32: 'tf.Tensor(1, shape=(), dtype=int32)'
```

```python
tf.summary.scalar('foo', 22, step=1.0)
...
TypeError: Cannot convert provided value to EagerTensor. Provided value: 1.0 Requested dtype: int64
```

**Comment by shashvatshahi1998** [comment]
pleasecan i contribute to this @nfelt. Can you guide me

**Comment by mohantym** [comment]
Hi @nfelt! This issue seems to be resolved in latest stable version TF 2.6 , Attaching [Gist](https://colab.research.google.com/gist/mohantym/c38f275170bae273e96bfb235abdfb0d/github_26407.ipynb) for reference.

**Comment by pjpratik** [comment]
I was able to reproduce this issue in TF Nightly 2.12.0.dev20221215. Please find the gist [here](https://colab.research.google.com/gist/pjpratik/0dc19a3b1f21d502ed6b61b26672b8a7/26407.ipynb). Thank you.
